pallets/settlement/src/lib.rs

```diff
@@ -602,6 +679,29 @@ decl_module! {
             Ok(Some(weight_for::weight_for_authorize_instruction::<T>() + weight_for_instruction_execution).into())
         }
 
+        /// Authorizes an existing confidential instruction.
+        ///
+        /// # Arguments
+        /// * `instruction_id` - Instruction id to authorize.
+        /// * `portfolios` - Portfolios that the sender controls and wants to authorize for this instruction
+        #[weight = weight_for::weight_for_authorize_instruction::<T>()
+            + weight_for::weight_for_transfer::<T>() // Maximum weight for `execute_instruction()`
+        ]
+        pub fn authorize_confidential_instruction(origin, instruction_id: u64, portfolios: BTreeSet<PortfolioId>, data: MercatTxData) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
```

Comment:
```suggestion
            let did = Identity::<T>::ensure_origin_call_permissions(origin)?.primary_did;
```
You need this new check in every extrinsic. It checks whether the caller has the permissions to call the extrinsic. `use Context` or `ensure_signed` should no longer be required. The bare `ensure_signed` doesn't check extrinsic permissions and should be avoided.

Reply:
@vkomenda I think permission changes are not present in the Mercat branch so it will be fine, We can leave a TODO

Reply:
@satyamakgec OK with that. @aafshar-poly Please feel free to resolve this unless you want to merge the latest develop into  this mercat.

Reply:
Seems like mercat branch has been updated. I'll address this once I rebase to the latest mercat branch.

Reply:
I don't see any usage of `primary_did` after rebasing. I am going to assume that this change is not in mercat branch yet.